onnx_caffe2/frontend.py

```diff
@@ -289,9 +289,21 @@ def caffe2_net_to_onnx_graph(net_def):
     graph_def.name = net_def.name
     # This is a mapping from Caffe2 names to ONNX names
     name_map = NameMap()
-    graph_def.input.extend(map(name_map.rename, net_def.external_input))
+    # TODO: This is cheating. Invent some magic to infer the types and
+    # shapes of graph input and output
+    graph_def.input.extend(
```

Comment:
Should we ask users for these shape and type info for caffe2_net?

Reply:
Yep, that's the straightforward (also my most preferred) way.

There is also discussion to export Caffe2 models in [MetaNetDef](https://github.com/caffe2/caffe2/blob/master/caffe2/proto/metanet.proto#L35) format.

I quickly post this hacky version to unblock pytorch side doing their fixes for the breaking change.

Reply:
In the future, I'd recommend emitting a warning when you exercise these code paths; it serves as a clear reminder that something *extremely* dodgy is happening until it gets excised ;)

Reply:
Yeah, will improve the error/warning message to help debugging in the future. For this one, I will commit a fix to clear this dodgy thing :-).